server/src/main/java/org/elasticsearch/index/mapper/GeoShapeIndexer.java

```diff
@@ -79,7 +79,10 @@ public Geometry prepareForIndexing(Geometry geometry) {
         return geometry.visit(new GeometryVisitor<>() {
             @Override
             public Geometry visit(Circle circle) {
-                throw new UnsupportedOperationException("CIRCLE geometry is not supported");
+                double[] latlon = new double[]{circle.getX(), circle.getY()};
+                normalizePoint(latlon);
+                double radius = circle.getRadiusMeters();
+                return new Circle(latlon[0], latlon[1], radius);
```

Comment:
I think this is wrong. GeoShapeIndexer works on geo_shape fields and CIRCLE is not a supported geometry for indexing.

Reply:
Sorry, I had misunderstood that, I've rolled back the change

Reply:
I understand now why you did this. I think we need to find a different solution, for example we should not be calling this class before creating the geo_shape query but another processor. Note that this processor is in charge for example in break a polygon when it crosses the dateline.

Reply:
I opened a PR to help this implementation

Reply:
PR Merged